components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/AuthenticationHandler.java

```diff
@@ -103,6 +106,18 @@ protected void postAuthenticate(MessageContext messageContext, AuthenticationRes
                         .getThreadLocalCarbonContext().getTenantDomain())) {
                     PrivilegedCarbonContext.getThreadLocalCarbonContext().setUsername(IdentityUtil.addDomainToName
                             (user.getUserName(), user.getUserStoreDomain()));
+                    try {
+                        if (user instanceof AuthenticatedUser) {
+                            PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                                    .setUserId(((AuthenticatedUser) user).getUserId());
+                        } else {
+                            AuthenticatedUser authenticatedUser = new AuthenticatedUser(user);
+                            PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                                    .setUserId(authenticatedUser.getUserId());
+                        }
+                    } catch (UserIdNotFoundException e) {
+                        LOG.error("User id not found for user: " + user.getLoggableUserId());
```

Comment:
If user id is not found what will come as the value for user.getLoggableUserId()

Reply:
If the user id is not available it will return the value from toFullQualifiedUsername() method invocation.